src/BinarySerializationTestForStructs.zig

```diff
@@ -0,0 +1,135 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const file = std.fs.File;
+const cwd = std.fs.cwd();
+const fba = std.heap.FixedBufferAllocator;
+
+const ArbitraryType = struct {
+    size: u64,
+    line: []u8
+};
+const BoxedStructure = packed struct {
+    flag: bool,
+    letter: u8,
+    u32_value: u32,
+};
+const arbitrary_output_name: []const u8 = "./output/arbitrary.bin";
+const boxed_type_one_output_name: []const u8 = "./output/boxed_type_one.bin";
+
+pub fn WriteArbitraryType(mem_alloc: std.mem.Allocator) !ArbitraryType {
+    var at: ArbitraryType = .{
+        .size = 25,
+        .line = try mem_alloc.alloc(u8, 25)
+    };
+    // Zero out the u8 array to ensure no trash values
+    // are written to the file.
+    @memset(at.line, 0);
+    const hello_world = "Hello, World!";
+    var ii: u64 = 0;
+    while (ii < hello_world.len) : (ii += 1) {
+        at.line[ii] = hello_world[ii];
+    }
```

Comment:
You can use `@memcpy` here, `@memcpy(at.line, hello_world);`, see https://ziglang.org/documentation/master/#memcpy for more information.